torch/_inductor/codegen/cpp_prefix.h

```diff
@@ -77,22 +92,33 @@ Welford<T> welford_combine(const Welford<T> &a, const Welford<T> &b) {
   auto result = Welford<T>{
     a.mean + delta * wb_over_w,
     a.m2 + b.m2 + delta * delta * a.weight * wb_over_w,
-    new_weight
+    new_weight,
+    a.index + b.index,
   };
   return result;
 }
 
 template <typename T>
-Welford<T> welford_combine(const Welford<T> &acc, T data) {
+Welford<T> welford_combine(const Welford<T> &acc, T data, const WeightRecp<T>* w=nullptr) {
   // Add a single data point
+  int64_t index = acc.index + 1;
   auto delta = data - acc.mean;
   auto new_weight = acc.weight + T(1);
-  auto new_mean = acc.mean + delta / new_weight;
+  T new_mean;
+  if constexpr (!IsVecType<T>::value) {
+    new_mean = acc.mean + delta / new_weight;
```

Comment:
I'm curious why you don't use `WeightRecp` for scalar loops as well?

Reply:
We only do this optimization for the vectorized part now, because only the tail part has scalar loop when generating the vec kernel, and there are not many scalar operations in the tail part. Of course, if it is a pure scalar loop, it may also be optimized for scalar, but this requires distinguishing between the pure scalar loop and tail scalar loop since the two cases require different lengths of weights. The same optimization may be done for scalar loop in future PRs.